src/README.md

```diff
@@ -13,12 +13,10 @@ This greatly reduces the amount of work needed to install and run the Po.et Node
 - Services communicate via pub/sub using [RabbitMQ], never directly.
 - Each service initializes and maintains its own mongo collections
 - Services must not access another services mongo collections. 
-- Classes are used as the main form of encapsulation and layered separation of concerns.
-- [Dependency Injection] is implemented using the [InversifyJS] framework. See [Inversion of Control]
+- Factory functions are used as the main form of encapsulation and layered separation of concerns.
+- [Dependency Injection] is implemented manually.
```

Comment:
`See [Inversion of Control]` would still apply... did you want to remove it?

Reply:
I think now that we are not using a framework it should be replaced with Dependency Inversion Principle https://en.wikipedia.org/wiki/Dependency_inversion_principle

Reply:
From the IoC wiki page:

> In software engineering, inversion of control (IoC) is a design principle in which custom-written portions of a computer program receive the flow of control from a generic framework. A software architecture with this design inverts control as compared to traditional procedural programming: in traditional programming, the custom code that expresses the purpose of the program calls into reusable libraries to take care of generic tasks, but with inversion of control, it is the framework that calls into the custom, or task-specific, code.

> The term is related to, but different from, the dependency inversion principle, which concerns itself with decoupling dependencies between high-level and low-level layers through shared abstractions

I actually was confusing these two terms thinking they were the same thing for the last few months until I looked into them again :S

Reply:
We are not exactly creating shared abstractions with our interfaces, we are sorta doing the half-assed version, but as you said before it's not truly necessary to abstract those interfaces.

Reply:
> We are not exactly creating shared abstractions with our interfaces, we are sorta doing the half-assed version, but as you said before it's not truly necessary to abstract those interfaces.

 Yup.

> I actually was confusing these two terms for the last few months, I was thinking they were the same thing. I just recently looked them up again and realized I was wrong. Let me know if I am still off haha smile

lol you got it right. the names can be confusing. I think IoC is a form of DIP, and DI is a subset of IoC... 

As always, software definitions are rather vague and sometimes no longer applicable, even though they are meant to stay true regardless the languages or paradigms. Modern DI Container libraries even are more library-ish than framework-ish. (where you manually set up the container and push stuff into it, rather than giving up the complete control of your application to a framework).

I argue that what we do is still Inversion of Control, and that we have a [composition root](http://blog.ploeh.dk/2011/07/28/CompositionRoot/), even though we use neither frameworks nor classes (soon™). 

Almost all pattern definitions scream "class" every three words... they need to be rewritten, but conceptually many still apply.

Also: patterns weren't really _defined_ by anybody. They were just observed, collected, worded and published by famous developers.